radvel/utils.py

```diff
@@ -542,12 +545,27 @@ def Msini(K, P, Mtotal, e, Msini_units='earth'):
     """
     # convert inputs to array so they work with units
     P = np.array(P)
-    Mtotal = np.array(Mtotal)
+    Mstar = np.array(Mstar)
     K = np.array(K)
     e = np.array(e)
-
-    P = (P * u.d).to(u.year).value
-    Msini = K / K_0 * np.sqrt(1.0 - e**2.0)*Mtotal**(2.0 / 3.0)*P**(1 / 3.0)
+    G = c.G.value                # added gravitational constant
+    Mjup = c.M_jup.value         # added Jupiter's mass
+    Msun = c.M_sun.value         # added sun's mass
+
+    P = (P * u.d).to(u.second).value
+    Mstar = Mstar*Msun
+       
+    a = K*(((2*(np.pi)*G)/P)**(-1/3))*np.sqrt(1-(e**2))
+    Msini = []
+    for i in range(len(P)):
```

Comment:
Can you get rid of this loop? This could end up being very slow for long MCMC chains.

Reply:
I thought MCMC doesn't call 'msini' function? This numeric solution can only take in one value at a time, instead of taking all the period values from MCMC chains at the same time. On my end MCMC works at the same speed as before but the 'derived' command is slower. But yes, this for loop was my last resolve as well. I'll definitely put it on my to do list if I have a better solution.

Reply:
Yeah, that's what I mean. It will just be slow to convert long MCMC chains into derived parameters.